src/sagemaker/predictor.py

```diff
@@ -105,11 +107,35 @@ def _create_request_args(self, data, initial_args=None):
         args['Body'] = data
         return args
 
-    def delete_endpoint(self):
-        """Delete the Amazon SageMaker endpoint backing this predictor.
+    def _delete_endpoint_config(self):
+        """Delete the Amazon SageMaker endpoint configuration
         """
+        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
+        endpoint_config_name = endpoint_description['EndpointConfigName']
+        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+
+    def delete_endpoint(self, delete_endpoint_config=True):
+        """Delete the Amazon SageMaker endpoint and endpoint configuration backing this predictor.
+
+        Args:
+            delete_endpoint_config (bool): Flag to indicate whether to delete the corresponding SageMaker endpoint
+                configuration tied to the endpoint. If False, only the endpoint will be deleted. (default: True)
+
+        """
+        if delete_endpoint_config:
+            self._delete_endpoint_config()
+
         self.sagemaker_session.delete_endpoint(self.endpoint)
 
+    def get_model(self):
```

Comment:
What will happen here if a user created the predictor outside of estimator or model.deploy().

It will return None I believe, and there isn't a nice way of getting the model.

Reply:
As we discussed offline, we will modify ``Predictor`` to delete the model directly, in align with ``Transformer``.